src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -808,6 +811,33 @@ private void paste() {
         }
     }
 
+    private void copyTitle() {
+        List<BibEntry> bes = mainTable.getSelectedEntries();
+        if (!bes.isEmpty()) {
+            storeCurrentEdit();
+            List<String> titles = new ArrayList<>(bes.size());
+
+            // Collect all non-null keys.
+            for (BibEntry be : bes) {
+                be.getTitle().ifPresent(titles::add);
```

Comment:
I've tried using a stream using your suggest now for retrieving the list of titles.
However, it required an additional mapping to be done in order to form the list of titles.